torchao/sparsity/README.md

```diff
@@ -0,0 +1,738 @@
+   <td>0
+   </td>
+   <td>0
+   </td>
+   <td>0
+   </td>
+   <td>1
+   </td>
+  </tr>
+</table>
+
+
+_Fig 2.3: unstructured sparsity_
+
+   </td>
+  </tr>
+  <tr>
+   <td>2:4 Semi-Structured
+
+   </td>
+   <td>
+
+<table>
+  <tr>
+   <td>0
+   </td>
+   <td>1
+   </td>
+   <td>1
+   </td>
+   <td>0
+   </td>
+   <td>1
+   </td>
+   <td>0
+   </td>
+   <td>1
+   </td>
+   <td>0
+   </td>
+  </tr>
+  <tr>
+   <td>0
+   </td>
+   <td>0
+   </td>
+   <td>1
+   </td>
+   <td>1
+   </td>
+   <td>1
+   </td>
+   <td>1
+   </td>
+   <td>0
+   </td>
+   <td>0
+   </td>
+  </tr>
+  <tr>
+   <td>1
+   </td>
+   <td>0
+   </td>
+   <td>0
+   </td>
+   <td>1
+   </td>
+   <td>0
+   </td>
+   <td>1
+   </td>
+   <td>0
+   </td>
+   <td>1
+   </td>
+  </tr>
+  <tr>
+   <td>0
+   </td>
+   <td>1
+   </td>
+   <td>0
+   </td>
+   <td>1
+   </td>
+   <td>1
+   </td>
+   <td>0
+   </td>
+   <td>1
+   </td>
+   <td>0
+   </td>
+  </tr>
+</table>
+
+
+_Fig 2.4: 2:4 semi-structured sparsity_
+
+   </td>
+  </tr>
+  <tr>
+   <td>Block Sparsity
+
+   </td>
+   <td>
+
+<table>
+  <tr>
+   <td>0
+   </td>
+   <td>0
+   </td>
+   <td>0
+   </td>
+   <td>0
+   </td>
+   <td>1
+   </td>
+   <td>1
+   </td>
+   <td>1
+   </td>
+   <td>1
+   </td>
+  </tr>
+  <tr>
+   <td>0
+   </td>
+   <td>0
+   </td>
+   <td>0
+   </td>
+   <td>0
+   </td>
+   <td>1
+   </td>
+   <td>1
+   </td>
+   <td>1
+   </td>
+   <td>1
+   </td>
+  </tr>
+  <tr>
+   <td>0
+   </td>
+   <td>0
+   </td>
+   <td>0
+   </td>
+   <td>0
+   </td>
+   <td>1
+   </td>
+   <td>1
+   </td>
+   <td>1
+   </td>
+   <td>1
+   </td>
+  </tr>
+  <tr>
+   <td>0
+   </td>
+   <td>0
+   </td>
+   <td>0
+   </td>
+   <td>0
+   </td>
+   <td>1
+   </td>
+   <td>1
+   </td>
+   <td>1
+   </td>
+   <td>1
+   </td>
+  </tr>
+</table>
+
+
+_Fig 2.5: 4x4 block-wise structured sparsity_
+
+   </td>
+  </tr>
+  <tr>
+   <td>Structured Sparsity
+
+   </td>
+   <td>
+
+<table>
+  <tr>
+   <td>1
+   </td>
+   <td>1
+   </td>
+   <td>1
+   </td>
+   <td>1
+   </td>
+   <td>1
+   </td>
+   <td>1
+   </td>
+   <td>1
+   </td>
+   <td>1
+   </td>
+  </tr>
+  <tr>
+   <td>0
+   </td>
+   <td>0
+   </td>
+   <td>0
+   </td>
+   <td>0
+   </td>
+   <td>0
+   </td>
+   <td>0
+   </td>
+   <td>0
+   </td>
+   <td>0
+   </td>
+  </tr>
+  <tr>
+   <td>1
+   </td>
+   <td>1
+   </td>
+   <td>1
+   </td>
+   <td>1
+   </td>
+   <td>1
+   </td>
+   <td>1
+   </td>
+   <td>1
+   </td>
+   <td>1
+   </td>
+  </tr>
+  <tr>
+   <td>0
+   </td>
+   <td>0
+   </td>
+   <td>0
+   </td>
+   <td>0
+   </td>
+   <td>0
+   </td>
+   <td>0
+   </td>
+   <td>0
+   </td>
+   <td>0
+   </td>
+  </tr>
+</table>
+
+
+_Fig 2.6: row-wise structured sparsity_
+
+   </td>
+  </tr>
+</table>
+
+
+_Table 4.4: Description of some common sparsity patterns. _
+
+
+# Design
+
+Pruning, like quantization, is an accuracy/performance trade-off, where we care not only about the speedup but also on the accuracy degradation of our architecture optimization technique.
```

Comment:
should we rename the folder to pruning

Reply:
I think sparsity is the more widely used term, so let's keep it called that. But I'll change Pruning -> Sparsity where it makes sense in the README